Explorer++/Helper/FileOperations.cpp

```diff
@@ -46,12 +46,24 @@ HRESULT FileOperations::RenameFile(IShellItem *item, const std::wstring &newName
 	return hr;
 }
 
-HRESULT FileOperations::DeleteFiles(HWND hwnd, const std::vector<PCIDLIST_ABSOLUTE> &pidls,
-	bool permanent, bool silent)
+HRESULT FileOperations::DeleteFiles(HWND hwnd, std::vector<PCIDLIST_ABSOLUTE> &pidls, bool permanent, bool silent)
 {
-	wil::com_ptr_nothrow<IFileOperation> fo;
-	HRESULT hr = CoCreateInstance(CLSID_FileOperation, nullptr, CLSCTX_ALL, IID_PPV_ARGS(&fo));
+	wil::com_ptr_nothrow<IFileOperation> fo{};
+	auto hr = CoCreateInstance(CLSID_FileOperation, nullptr, CLSCTX_ALL, IID_PPV_ARGS(&fo));
+	if (FAILED(hr) || !fo)
+	{
+		return hr;
+	}
+
+	wil::com_ptr_nothrow<IFileOperationProgressSink> pSink{};

```

Comment:
It would be better to not include any mention of the type in variable names like this. That is, this should just be `sink`, rather than `pSink`. Some older code still uses Hungarian notation, but for quite a while now, newer code has used plain names.